src/server/index.js

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable no-console */
+const WebSocket = require('ws');
+const { v4: uuidv4 } = require('uuid');
+
+const wss = new WebSocket.Server({ port: 8080 });
+
+const rooms = {};
+
+function broadcast(roomId, data) {
+  const room = rooms[roomId];
+
+  if (!room) {
+    return;
+  }
+
+  room.users.forEach((user) => {
+    if (user.ws.readyState === WebSocket.OPEN) {
+      user.ws.send(JSON.stringify(data));
+    }
+  });
+}
+
+wss.on('connection', (ws) => {
+  let currentRoom = null;
+
+  ws.on('message', (message) => {
+    const data = JSON.parse(message);
+
+    switch (data.type) {
+      case 'setUsername':
+        ws.username = data.username;
+        ws.color = `hsl(${Math.random() * 360}, 100%, 50%)`;
+        break;
+
+      case 'createRoom':
+        const roomId = uuidv4();
+
+        rooms[roomId] = { name: data.roomName, messages: [], users: [] };
+        ws.send(JSON.stringify({ type: 'roomCreated', roomId }));
+        break;
+
+      case 'joinRoom':
+        if (currentRoom && rooms[currentRoom]) {
+          rooms[currentRoom].users = rooms[currentRoom].users.filter(
+            (user) => user.ws !== ws,
+          );
+        }
+        currentRoom = data.roomId;
+
+        if (!rooms[currentRoom]) {
+          return;
```

Comment:
When a user tries to join a non-existent room, consider sending a response back to the client to inform them that the room does not exist. This will improve user feedback and experience.